src/client.ts

```diff
@@ -171,6 +186,50 @@ export default class Client {
     );
   }
 
+  private async setupCustomGemfile() {
+    // If we're working on the ruby-lsp itself, we can't create a custom Gemfile or we'd be trying to activate the same
+    // gem twice
+    if (this.workingFolder.endsWith("ruby-lsp")) {
+      return;
+    }
+
+    // Create the .ruby-lsp directory if it doesn't exist
+    const rubyLspDirectory = path.join(this.workingFolder, ".ruby-lsp");
+
+    if (!fs.existsSync(rubyLspDirectory)) {
+      fs.mkdirSync(rubyLspDirectory);
+    }
+
+    // Ignore the .ruby-lsp directory automatically
+    const gitIgnorePath = path.join(rubyLspDirectory, ".gitignore");
+
+    if (!fs.existsSync(gitIgnorePath)) {
+      fs.writeFileSync(gitIgnorePath, "*");
+    }
```

Comment:
Since the `.gitignore` is checked inside the repo doesn't git tells you you have pending changes because the `.gitignore` has been modified?

Reply:
That's not what's happening though. We create a `.gitignore` file inside of the `.ruby-lsp` folder and use `*` to ignore the entire thing.

No changes are applied to the top level `.gitignore`, there are no git modifications after this runs.

Reply:
Ahh okay so you mark all the files as `.gitignore` in the sub directory which make the main directory not show the `ruby-lsp` at all? Very nice!

Reply:
I still only have a purist bias towards `.git/info/exclude` because this seems to fit the exact use-case for that, as I quoted from the documentation:
> ... (e.g., auxiliary files that live inside the repository but are specific to one user’s workflow) should go into the $GIT_DIR/info/exclude file.

Having said that, this is not an objection, just stating my bias towards that.

Reply:
I would also like to add that adding to `.git/info/exclude` should be more resilient for cases where one might remove the `.gitignore` file inside the `.ruby-lsp` folder by mistake. I have yet not seen anyone remove something in the `.git` folder by mistake. Again, not a huge deal.

Reply:
I agree, we should use the right tool for the job and it seems that git already prescribes the exclude file. I changed the code to use this strategy.